LX/src/main/java/heronarts/lx/LXSnapshotController.java

```diff
@@ -0,0 +1,161 @@
+package heronarts.lx;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import heronarts.lx.osc.OscMessage;
+
+public class LXSnapshotController extends LXComponent {
+
+  private Integer nextShapshotID = 1;
+
+  private final List<LXSnapshot> snapshots = new ArrayList<LXSnapshot>();
+  private final TreeMap<Integer, LXSnapshot> snapshotsDict = new TreeMap<Integer, LXSnapshot>();
+
+  public interface SnapshotListener {
```

Comment:
Fine to just call this Listener, so the full class name is LXSnapshotEngine.Listener